lms/djangoapps/courseware/user_state_client.py

```diff
@@ -239,10 +286,19 @@ def set_many(self, username, block_keys_to_state, scope=Scope.user_state):
                     log.warning("set_many: All {} block keys: {}".format(
                         len(block_keys_to_state), block_keys_to_state.keys()
                     ))
+                else:
+                    # The state of an existing block was modified, and no
+                    # exception was raised.  Collect statistics for NR metrics.
+                    self._nr_block_count_increment('set_many', usage_key.block_type)
+                    self._nr_block_size_accumulate('set_many', usage_key.block_type, len(student_module.state))
+            else:
+                # A new block was created, so collect statistics for NR
+                # metrics.
+                self._nr_block_count_increment('set_many', usage_key.block_type)
+                self._nr_block_size_accumulate('set_many', usage_key.block_type, len(student_module.state))
```

Comment:
Nit: You could move these down a few lines below where the datadog reporting for `if created:` is.